### PR TITLE
Android WorkManager.md

### PR DESCRIPTION
**Pull request by Muddz**
First release. Not finished yet

### CODE REVIEW

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+
+#### 2) The second Worker is the one which add some weather and location stickers on the image
+```java
+class AddStickersTask extends Worker {
+   @NonNull
+   @Override
+   public WorkerResult doWork() {
+       if(IsGpsEnabled()){
+       return WorkerResult.SUCCESS;
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: Fix the indention here

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+
+GPS requirement is not yet supported in the `Constraints` class but we will instead check for enabled GPS in the `AddStickersTask` and if it’s not enabled we will return `FAILURE` and the next `WorkManager` won’t proceed to the next `Worker`
+
+### Putting everything together 
+
+Now lets create new instances of our 3 `Worker` classes: 
+
+```java
+OneTimeWorkRequest imageCompressionTask = new OneTimeWorkRequest.Builder(ImageCompressionTask.class).build();
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: This doesnt look great in markdown, can you add line breaks after "="?

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+   public ImageCompressionTask(Bitmap bitmap) {
+       this.bitmap = bitmap;
+   }
+
+   @NonNull
+   @Override
+   public WorkerResult doWork() {
+       Bitmap newBitmap;
+       try {
+           newBitmap = Bitmap.createScaledBitmap(bitmap, 500, 500, false);
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: shouldn't we do something with the bitmap? I know it's proof of concept, but still :D

Comment (johsoe) [reply source/_posts/2018-08-06-Android WorkManager.md]: At least add in a comment with `// Write new bitmap to persistence` or something

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        }
+    }
+```
+
+
+#### 3) In the last Worker we just upload our Bitmap to our server
+```kotlin
+    class UploadImageTask : Worker() {
+        override fun doWork(): WorkerResult {
+            //Retrive bitmap and upload work here
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: *Retrieve

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+#### 1) The first Worker compress our Bitmap into a smaller size, convert the Bitmap to ByteArray passes it in the WorkManager's outputData object and then we return WorkerResult.SUCCESS or WorkerResult.FAILURE.
+
+```kotlin
+    class ImageCompressionTask(val bitmap: Bitmap?) : Worker() {
+        override fun doWork(): WorkerResult {
+            val newBitmap: Bitmap?
+            try {
+                //Create a compressed bitmap
+                newBitmap = Bitmap.createScaledBitmap(bitmap, 500, 500, false)
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: Does this work? Doesnt it complain about nullable bitmap?

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                return WorkerResult.SUCCESS
+            } catch (e: IllegalArgumentException) {
+                return WorkerResult.FAILURE
+            }
+        }
+    }
+```
+
+
+#### 2) In the second Worker we retrive the Bitmap from the Workers inputData object and adds some weather and location stickers on the image
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: *Retrieve

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+```kotlin
+    class AddStickersTask : Worker() {
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: Can we add something about WorkerResult.RETRY? Maybe if it failed to get a location?

File: source/_posts/2018-08-06-Android WorkManager.md (hunk 6 above)
Comment (Muddz) [reply source/_posts/2018-08-06-Android WorkManager.md]: Fixed

File: source/_posts/2018-08-06-Android WorkManager.md (hunk 4 above)
Comment (Muddz) [reply source/_posts/2018-08-06-Android WorkManager.md]: Fixed

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+GPS requirement is not yet supported in the `Constraints` class but we will instead check for enabled GPS in the `AddStickersTask` and if it’s not enabled we will return `FAILURE` and the next `WorkManager` won’t proceed to the next `Worker`
+
+
+#### Now lets use our Worker classes
+
+Here we set our `Constraints` for each `Worker` and `build()` will return an instance of `OneTimeWorkRequest`
+
+```kotlin
+val imageCompressionTask = OneTimeWorkRequest.Builder(ImageCompressionTask::class.java).build()
+val addStickersTask = OneTimeWorkRequest.Builder(AddStickersTask::class.java).setConstraints(constraint).build()
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: Add some linebreaks here so we avoid horizontal scrolling

File: source/_posts/2018-08-06-Android WorkManager.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+
+#### 2) In the second Worker we retrieve the Bitmap from the Workers inputData object and adds some weather and location stickers on the image
+```kotlin
+    class AddStickersTask : Worker() {
+        override fun doWork(): WorkerResult {
+            try {
+                //Adding stickers on the bitmap...
+                return WorkerResult.SUCCESS
```
Comment (johsoe) [inline source/_posts/2018-08-06-Android WorkManager.md]: Maybe add something about WorkerResult.RETRY here?

**Pull request merged**